.pre-commit-config.yaml

```diff
@@ -0,0 +1,10 @@
+repos:
+  - repo: https://github.com/psf/black
+    rev: 22.12.0
+    hooks:
+      - id: black
```

Comment:
If there are changes a pre-commit script makes, it fails the commit but also updates the files. Then you have to notice that and re-commit.

Could we instead add a flag somewhere where it will update the files and then do the commit anyway? That seems harmless to me.

Reply:
precommit explicitly rejects doing this - see https://stackoverflow.com/a/58459499/5318482

Reply:
So OK. I understand why the author of the tool would have an ideological reason against it in general.

But in this case (`black`), what are we meaningfully supposed to do in response to a failure? Can you honestly say you'll review the changes and be like 'yup' that's good? -- if `black` were to show the diff it wanted to do (or be in dry run mode), maybe you could have a meaningful response. But by the time you get the failure, `black` has already run. So... you better be OK with it.

Which I am! We trust `black`. So why abort the commit?

But if you are not down for the work around the `pre-commit` author offers of `git add -u' --`, my other concern is more about visibility of a failure:

------------------------------------------------------
$ git commit -a -m WIP
black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted ctakesclient/__init__.py

All done! ✨ 🍰 ✨
1 file reformatted.
------------------------------------------------------

Us humans are imperfect (hence `black`) so I don't think it's crazy that you run the commit see the shiny `All done!` and be like cool, then move on / push / edit more, before realizing nope - nothing actually happened there.

So if you don't like committing anyway on modified files, maybe there's room for making the failure case super super obvious?

(Or maybe this is just Mike things, not being used to pre-commits and not used to looking for the Failed word all the time. This is my first time with meaningful `pre-commits` and this is a significant work flow change for me and maybe I'm just grumpy.)

Reply:
(I'm fine with trying it for a bit and seeing if I do in fact miss the "modified files" message.)

Reply:
But I want to see if you're swayable on auto-modification first. If not, OK.

Reply:
like - you can talk me into it if you're cool with it.